src/Symfony/Component/Config/ResourceCheckerConfigCache.php

```diff
@@ -118,12 +118,24 @@ public function write(string $content, ?array $metadata = null): void
         }
 
         if (null !== $metadata) {
-            $filesystem->dumpFile($this->metaFile, serialize($metadata));
+            $filesystem->dumpFile($this->metaFile, $ser = serialize($metadata));
             try {
                 $filesystem->chmod($this->metaFile, $mode, $umask);
             } catch (IOException) {
                 // discard chmod failure (some filesystem may not support it)
             }
+
+            $ser = preg_replace_callback('/;O:(\d+):"/', static fn ($m) => ';O:'.(9 + $m[1]).':"Tracking\\', $ser);
+            $ser = preg_replace_callback('/s:(\d+):"\0[^\0]++\0/', static fn ($m) => 's:'.($m[1] - \strlen($m[0]) + 6).':"', $ser);
+            $ser = json_encode(unserialize($ser));
+            $ser = str_replace('"__PHP_Incomplete_Class_Name":"Tracking\\\\', '"@type":"', $ser);
```

Comment:
This logic should clearly be covered by tests

Reply:
test added